### PR TITLE
gateway: pprof endpoint

### PR DESCRIPTION
**Pull request by mcutalo88**
### Description
Adding the ability to enable pprof debug endpoints if configured.

### Testing Performed
Locally.

```
gateway:
  enable_pprof: true
```

```
➜ ~ curl -sK -v http://localhost:8080/debug/pprof/heap > heap.out

➜ ~ go tool pprof heap.out
Type: inuse_space
Time: Dec 2, 2020 at 1:28pm (PST)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) png
Generating report in profile002.png
```

Attached is the result
profile002

### CODE REVIEW

File: backend/gateway/mux/pprof.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package mux
+
+import (
+	"net/http"
+	"net/http/pprof"
+)
+
+func pprofHandlers(httpMux *http.ServeMux) {
+	httpMux.Handle("/debug/pprof/", http.HandlerFunc(pprof.Index))
```
Comment (danielhochman) [inline backend/gateway/mux/pprof.go:9]:
```suggestion
```

File: backend/gateway/mux/pprof.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	httpMux.Handle("/debug/pprof", http.HandlerFunc(pprof.Index))
```
Comment (danielhochman) [inline backend/gateway/mux/pprof.go:10]:
```suggestion
	httpMux.HandleFunc("/debug/pprof", pprof.Index)
```

same for others

File: backend/gateway/mux/pprof.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	httpMux.Handle("/debug/pprof/cmdline", http.HandlerFunc(pprof.Cmdline))
+	httpMux.Handle("/debug/pprof/profile", http.HandlerFunc(pprof.Profile))
+	httpMux.Handle("/debug/pprof/symbol", http.HandlerFunc(pprof.Symbol))
+	httpMux.Handle("/debug/pprof/trace", http.HandlerFunc(pprof.Trace))
+	httpMux.Handle("/debug/goroutine", pprof.Handler("goroutine"))
+	httpMux.Handle("/debug/heap", pprof.Handler("heap"))
```
Comment (danielhochman) [inline backend/gateway/mux/pprof.go:16]: index already handles the other profiles from what i can tell 
https://github.com/golang/go/blob/master/src/net/http/pprof/pprof.go#L380-L396
https://github.com/golang/go/blob/dda2991c2ea0c5914714469c4defc2562a907230/src/runtime/pprof/pprof.go#L186-L193

Comment (mcutalo88) [reply backend/gateway/mux/pprof.go:16]: let me give it a try

File: backend/gateway/mux/pprof.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package mux
+
+import (
+	"net/http"
+	"net/http/pprof"
+)
+
+func pprofHandlers(httpMux *http.ServeMux) {
+	httpMux.HandleFunc("/debug/pprof/", pprof.Index)
```
Comment (danielhochman) [inline backend/gateway/mux/pprof.go:9]: let's just fold this inline in mux.go, no need for the indirection to this new file

**Review by danielhochman — APPROVED** [review]

File: backend/gateway/mux/pprof.go (hunk 4 above)
Comment (mcutalo88) [reply backend/gateway/mux/pprof.go:9]: yeah sure

**Pull request merged**